service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -0,0 +1,305 @@
+ */
+public abstract class BaseTenantService<T> extends ConfigurationSupportingVerticle<T> implements TenantService {
+
+    /**
+     * A logger to be shared by subclasses.
+     */
+    protected final Logger log = LoggerFactory.getLogger(this.getClass());
+    private MessageConsumer<JsonObject> tenantConsumer;
+
+
+    /**
+     * Starts up this service.
+     * <ol>
+     * <li>Registers an event bus consumer for address {@link TenantConstants#EVENT_BUS_ADDRESS_TENANT_IN} listening for
+     * tenant CRUD requests.</li>
+     * <li>Invokes {@link #doStart(Future)}.</li>
+     * </ol>
+     *
+     * @param startFuture The future to complete on successful startup.
+     */
+    public final void start(final Future<Void> startFuture) {
+        this.tenantConsumer();
+        this.doStart(startFuture);
+    }
+
+    /**
+     * Subclasses should override this method to perform any work required on start-up of this verticle.
+     * <p>
+     * This method is invoked by {@link #start()} as part of the verticle deployment process.
+     * </p>
+     *
+     * @param startFuture future to invoke once start up is complete.
+     */
+    protected void doStart(final Future<Void> startFuture) {
+        // should be overridden by subclasses
+        startFuture.complete();
+    }
+
+    private void tenantConsumer() {
+        this.tenantConsumer = this.vertx.eventBus().consumer(TenantConstants.EVENT_BUS_ADDRESS_TENANT_IN);
+        this.tenantConsumer.handler(this::processTenantMessage);
+        this.log.info("listening on event bus [address: {}] for incoming tenant CRUD messages",
+                TenantConstants.EVENT_BUS_ADDRESS_TENANT_IN);
+    }
+
+    /**
+     * Unregisters the registration message consumer from the Vert.x event bus and then invokes {@link #doStop(Future)}.
+     *
+     * @param stopFuture the future to invoke once shutdown is complete.
+     */
+    public final void stop(final Future<Void> stopFuture) {
+        this.tenantConsumer.unregister();
+        this.log.info("unregistered tenant data consumer from event bus");
+        this.doStop(stopFuture);
+    }
+
+    /**
+     * Subclasses should override this method to perform any work required before shutting down this verticle.
+     * <p>
+     * This method is invoked by {@link #stop()} as part of the verticle deployment process.
+     * </p>
+     *
+     * @param stopFuture the future to invoke once shutdown is complete.
+     */
+    protected void doStop(final Future<Void> stopFuture) {
+        // to be overridden by subclasses
+        stopFuture.complete();
+    }
+
+    /**
+     * Processes a tenant request message received via the Vertx event bus.
+     *
+     * @param tenantMsg The message.
+     */
+    public final void processTenantMessage(final Message<JsonObject> tenantMsg) {
+        try {
+            final JsonObject body = tenantMsg.body();
+            final String tenantId = body.getString(RequestResponseApiConstants.FIELD_TENANT_ID);
```

Comment:
Since we are implementing the tenant service we should refer to all constants via `TenantConstants` IMHO, thus not relying on the fact that the (shared) constants are actually defined in a common superclass.

Reply:
Full ack :+1: